test/assertions.js

```diff
@@ -1,19 +1,64 @@
+import { curry, type, cond, and, map, zip, T, equals, mapObjIndexed } from 'ramda';
+import expect from 'expect.js';
+import { List } from 'immutable';
+
 import read from '../src/reader/token-reader';
-import { compile } from '../src/sweet';
+import { compile, parse } from '../src/sweet';
 import { Enforester } from '../src/enforester';
-import { List } from 'immutable';
 import StoreLoader from '../src/store-loader';
 
 export const stmt = x => x.items[0];
 export const expr = x => stmt(x).expression;
 export const items = x => x.items;
 
-
 export function makeEnforester(code) {
   let stxl = read(code);
   return new Enforester(stxl, List(), {});
 }
 
+function checkObjects(expected, actual) {
+  const isString = (x) => type(x) === 'String';
+  const isObject = (x) => type(x) === 'Object';
+  const isArray = (x) => type(x) === 'Array';
+
+  const checkWithHygiene = cond([
+    [and(isString, equals('<<hygiene>>')), curry((a, b) => true)],
+    [isObject, curry((a, b) => checkObjects(a, b))],
+    [isArray, curry((a, b) => (expect(b).to.have.length(a.length), map(([a, b]) => checkObjects(a, b), zip(a, b))))],
+    [T, curry((a, b) => expect(b).to.be(a))]
+  ]);
+
+  mapObjIndexed((prop, key, ob) => {
+    const checker = checkWithHygiene(prop);
+    expect(actual).to.have.property(key);
+    checker(actual[key]);
+  }, expected);
+}
+
+function getAst(code) {
+  const store = new Map();
+  store.set('main.js', code);
+
+  const loader = new StoreLoader(__dirname, store);
+  return parse('main.js', loader);
+}
+
+function testParseWithOpts(acc, code, expectedAst) {
+  try {
+    checkObjects(expectedAst, acc(getAst(code)));
+  } catch (e) {
+    throw new Error(e.message);
+  }
+}
+
+export function testParse(t, acc, code, expectedAst) {
+  return testParseWithOpts(acc, code, expectedAst);
+}
+
+export function testParseComparison(t, acc, codeA, codeB) {
+  testParse(t, acc, codeA, acc(getAst(codeB)));
```

Comment:
I think you should just use `t.deepEqual` for this kind of tests and avoid the custom "object equality modulo `<<hygiene>>` string". The tests you added don't require the custom stuff and I would prefer if we could avoid reintroducing tests that assert on hand written ASTs.